src/xpu/sycl/types.hpp

```diff
@@ -146,6 +177,23 @@ struct md_t {
 #undef CHECK_AND_ASSIGN
     }
 
+    // This constructor is to be called inside the kernel
```

Comment:
How do you control this copy constructor will be called only from the inside the kernel and not from other places?